spec/services/user_updater_spec.rb

```diff
@@ -46,6 +46,42 @@
       expect(user.reload.name).to eq "Jim Tom"
     end
 
+    describe "the user_updater_commit_updates event" do
+      it "allows plugins to perform additional updates" do
+        update_attributes = { name: "Jimmmy Johnny" }
+        handler =
+          Proc.new do |user, attrs|
+            user.user_profile.update!(bio_raw: "hello world I'm Jimmmy")
+            expect(attrs).to eq(update_attributes)
+          end
```

Comment:
You can use `DiscourseEvent.track_events` to ensure that the event is triggered with the right parameters.

Reply:
The thing is I want to test that updates made inside the event handler are actually persisted. How would I test that with `track_events` (and without registering a custom handler)?

Reply:
That would be a plugin's concern wouldn't it?

Reply:
Hmm tricky 🤔 If core promises that the event will be triggered before the `UPDATE` transaction is committed, then I feel like core tests should catch regressions when the event is no longer triggered inside the transaction.

Reply:
Ahh ok I missed that part where the event is triggered within the transaction. I think the test case is fine but maybe `within_user_updater_transaction` is a better even name to reflect that the events is triggered within a transacction?

Reply:
Done! ✅